src/main/java/de/hysky/skyblocker/skyblock/quicknav/QuickNavButton.java

```diff
@@ -42,13 +49,23 @@ public boolean toggled() {
      * @param toggled the toggled state of the button.
      * @param command the command to execute when the button is clicked.
      * @param icon    the icon to display on the button.
+     * @param tooltip the tooltip to show when hovered
      */
-    public QuickNavButton(int index, boolean toggled, String command, ItemStack icon) {
+    public QuickNavButton(int index, boolean toggled, String command, ItemStack icon, String tooltip) {
         super(0, 0, 26, 32, Text.empty());
         this.index = index;
         this.toggled = toggled;
         this.command = command;
         this.icon = icon;
+        if (tooltip == null || tooltip.isEmpty()) return;
+        if (tooltip.startsWith("{") || tooltip.startsWith("[")) {
+            TextCodecs.CODEC.decode(JsonOps.INSTANCE, SkyblockerMod.GSON.fromJson(tooltip, JsonElement.class))
+                    .ifError(pairError -> QuickNav.LOGGER.error("[Skyblocker] Failed to parse quicknav error for index {}. {}", index, pairError.message()))
+                    .ifSuccess(textJsonElementPair -> setTooltip(Tooltip.of(textJsonElementPair.getFirst())));
+
+
+        } else setTooltip(Tooltip.of(Text.literal(tooltip)));
```

Comment:
You can just display the literal text if there's an error, instead of checking if it starts with a bracket.

Reply:
don't wanna send an error when they just want a literal, unless i send a literal only when it starts with { or [

Reply:
You don’t send any errors. I’m saying try parsing the text as json and if it fails, just display the raw string literal. So users would see the json string if it’s malformed, and they can also type in whatever string they want.

Reply:
QuickNav.LOGGER.error() is there tho? unless it doesn't get called for some dubious reason

Reply:
Something like this.
```suggestion
        try {
            setTooltip(Tooltip.of(TextCodecs.CODEC.decode(JsonOps.INSTANCE, SkyblockerMod.GSON.fromJson(tooltip, JsonElement.class)).getOrThrow().getFirst()));
        } catch (Exception e) {
            setTooltip(Tooltip.of(Text.literal(tooltip)));
        }
```

Reply:
but there's no error logging :(

Reply:
sure